### ISSUE TITLE
processingLog has times in wrong zone

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1564]
See a comment in #1563

**Comment by dankelley** [comment]
Commit 3a9bce25c4dd6772313f216e980d67bfd87cbc53 of the "develop" branch now works for base `oce` objects and for `ctd` objects, as demonstrated below. The `oce` code uses a one-expression syntax (click the link to see the diff) that I don't think most users will understand, and we don't need that brevity in derived objects, so the `ctd` method is a two-step process.

I will code the two-step process for the other objects sometime in the next few days, and close the issue when I'm done.

(PS. I can also base a one-expression scheme on `.POSIXct()`, but I am of the impression that it is not recommended for use outside R itself.)


```R
> Sys.time()
[1] "2019-05-23 18:55:54.164 ADT"
> new('oce')[["processingLog"]]
$time
[1] "2019-05-23 21:54:00.347 UTC"

$value
[1] "Create oce object"

> new('ctd')[["processingLog"]]
$time
[1] "2019-05-23 21:55:57.817 UTC"

$value
[1] "create 'ctd' object"
```

**Comment by dankelley** [comment]
What the heck??? Look at the times below (run as a script so no typing time). How do we go from 19:05 ADT to 21:54 UTC?  Shouldn't the second be 21:06 for a 3hour shift?  This is very weird -- I'll look tomorrow.
```R
> library(oce)
Loading required package: testthat
Loading required package: gsw
> Sys.time()
[1] "2019-05-23 19:05:33.875 ADT"
> new('oce')[["processingLog"]]
$time
[1] "2019-05-23 21:54:00.347 UTC"

$value
[1] "Create oce object"

> new('ctd')[["processingLog"]]
$time
[1] "2019-05-23 22:05:33.938 UTC"

$value
[1] "create 'ctd' object"

```

**Comment by dankelley** [comment]
I'll try
```R
> {t<-Sys.time();attr(t,"tzone")<-"UTC";t}
[1] "2019-05-23 22:15:01.032 UTC"
> Sys.time()
[1] "2019-05-23 19:15:05.631 ADT"
```

**Comment by dankelley** [comment]
The weird timezone error (differences not being multiples of 1hour or even 1/2 hour) occurs only on my home machine (latest macos beta version) and not on my work machine (years-old macos).  Also, I checked the docs on `attr<-`, and the *setting* style is not documented to return anything. Since I get reasonable results with the multistep approach (as in the just-previous comment), I am going to use that, in both the base `oce` class and in all derived classes. I will also propagate that to the vignette, and explain it there.

**Comment by dankelley** [comment]
Actually, I don't see any harm in using `lubridate::now(tz="UTC")`.  We already "suggest" lubridate, and all this would mean is "requiring" it. That's a slight hassle, because it means users will have to install that package. But it is a *very* useful package, and it provides a lot of things that `oce` users would likely appreciate.  Indeed, `now()` is a perfect case in point -- that is a **lot** easier to remember than `Sys.time()` which I always guess is `sys.time()` or `Sys.Time()`.

I am going to code this in the next hour. If I get some screams from others about my approach, I can uncode it on the weekend.

**Comment by dankelley** [comment]
Hm, cancel that. I think `lubridate` is great and all, but if I add it to the `Depends` clause in `DESCRIPTION` and build `oce`, then I get a whole slew of new documentation files, for accessors to objects. These are all ones with `ANY` in the arglist, so maybe in some theoretical sense they ought to be there, but they will be so generic as to not be too helpful, I think. 

Previously, to get an overview you would use `"[[,oce-method"` and to get for ctd you'd use `"[[,ctd-method"`. I like that simple scheme, and don't see that users will be served by having yet another suite of a dozen or so help files that are at the generic form.  (I won't go into details here of what those files contain ... I just don't want the `see other` lists getting doubled in size.)

The `t<-Sys.time();attr(t,"tzone")<-"UTC";t}` method works fine, but it *is* a bit of a hassle. I wonder whether `oce` ought to have something like `clock(tz="UTC")` that returns the present time in the stated timezone?  To my surprise, "clock" is not in the namespace of an R console that has `oce` loaded.  And I think a lot of `oce` users would like to be able to use `clock()` to get the present time in UTC.

So, *that* is what I'm going to do.

**Comment by dankelley** [comment]
I chose `presentTime()` as the name for the new function.  I've gone through the whole code to find instances of using `Sys.time()` (thanks, grep!) and the build-test-check process works well on my work machine. I just pushed to github (branch "develop", commit aeda606bbcbc6f3054b386cf50884551fd7f84f0) so in about 40 minutes I'll know whether the travisCI build worked OK. If it did, then I'll close this.
